cdxev/merge.py

```diff
@@ -18,7 +18,76 @@
 logger = logging.getLogger(__name__)
 
 
-def merge_components(governing_sbom: dict, sbom_to_be_merged: dict) -> t.List[dict]:
+def filter_component(
+    present_components: list[ComponentIdentity],
+    new_components: list,
+    kept_components: list,
+    dropped_components: list,
+    add_to_existing: dict,
+) -> list[dict]:
+    """
+    Function that goes through a list of components and their nested sub components
+    and determine if they are present in a provided list with component identities.
+
+    The function operates directly on the lists and dictionary provided and returns
+    a list of filtered top level components that were not found in present_components.
+    Filtered means, that the nested components are also not already present.
+
+    param present_components: a list of component identities that are already present in the sbom.
+    param new_components: a list of components that shell be compared against the list of already
+    present components.
+    param kept_components: list of components not present in the list of provided components,
+                           including nested components
+    param dropped_components: list of components that are already present
+    param add_to_existing: list of nested components that have to be added to present components
```

Comment:
dot at end of the sentence + should be either present_components or adjust the present_components to present_components in [L#33](https://github.com/Festo-se/cyclonedx-editor-validator/pull/338/files#diff-9b28318ee8ce015ce5d42609086ae25efa5d528b3fcb84ee73e1014f3fe815dfR33).